### PR TITLE
Cleanup dependencies

### PR DESCRIPTION
**Pull request by erandagan**
- Removed unused dev dependencies
- Removed `react-native` from `devDependencies` (related to #156)
- Updated lockfile (fixes current CI errors)

### CODE REVIEW

**Comment by Trancever** [comment]
@erandagan Cn you fix CI? There are two indentation problems, shouldn't take much time :)

**Comment by erandagan** [comment]
> @erandagan Cn you fix CI? There are two indentation problems, shouldn't take much time :)

I did fix some errors in a separate branch, I can merge that into this one or open a new PR :)

**Comment by Trancever** [comment]
@erandagan Let's open separate PR.

**Comment by erandagan** [comment]
#160

File: package.json [hunk 1]
```diff
@@ -18,22 +18,14 @@
...
     "fbemitter": "^2.1.1"
   },
   "devDependencies": {
-    "@callstack/eslint-config": "^2.0.0",
     "@types/fbemitter": "^2.0.32",
     "@types/react-native": "^0.57.4",
-    "eslint": "^5.6.1",
-    "eslint-plugin-react-native": "^3.3.0",
-    "flow-bin": "~0.67.1",
-    "react": "16.3.1",
```
Comment (Trancever) [inline package.json]: Is it correct? We import React and React Native in many files and it's going to fail in dev mode when we remove it from devDependencies.

File: package.json [hunk 2]
```diff
@@ -18,22 +18,14 @@
   "author": "Suraj Tiwari <surajtiwari020@gmail.com>",
   "license": "MIT",
   "peerDependencies": {
-    "react": "*",
-    "react-native": "*"
+    "react-native": "^0.57.4"
```
Comment (Trancever) [inline package.json]: Any reason behind 0.57.4? Many people still use 0.55.x, it would be good to support them also.

File: package.json (hunk 1 above)
Comment (Trancever) [reply package.json]: Please reinstall node_modules locally and try to run example app. I can't see nor React-Native neither React in node_modules.

Comment (erandagan) [reply package.json]:
```
cd example
rm -rf node_modules
yarn
file react-native
> react-native: directory
```

Seems like it's there :)

Comment (Trancever) [reply package.json]: Ok my bad, i was trying to run example app not from example directory :D

**Pull request merged**